### DISCUSSION TITLE
Tableview autofit_columns does not work with changed fonts.

### DISCUSSION THREAD

**Discussion by Iamrodos** [israel-dryer/ttkbootstrap#399] [Q&A] [answered]
I am new to Tk and ttkbootstrap so I don't know if what I am experiencing is expected or if there is a way to improve the default behaviour.

If you change the font used for the Tableview, the calculations for autofit_columns do not work as it uses TkDefaultFont to calculate its measurements. 

The font used within autofit_columns is
`f = font.nametofont("TkDefaultFont")`

and it calculates the width via the .measure function
`new_width = f.measure(str(value)) + pad`

I am changing the font for the Tableview by using the technique in discussion "[Changing font for all ttkbootstrap widgets #322](https://github.com/israel-dryer/ttkbootstrap/discussions/322)". I set the font before the Tableview is created. Here is what I am doing.
`ttk.Style().configure(".", font=("Cascadia Code", 14))`

I created a workaround by doing essentially the same thing as autofit_columns by calculating my own width based on the modified font and setting the column width directly. Like this
```
    # Autofit of the columns does not work if not using the default font.
    # Emulate what Autofit does outselves
    f = font.Font(font=("Cascadia Code", 14))
    # Get the max column with for all of the rows
    maxes = [max(f.measure(row[key]) + f.measure(" ") for row in results) for key in results[0].keys()]
    # Apply the max with to each column
    for i, len in enumerate(maxes):
        table.view.column(i, width=len)
```
Pleased that I have a workaround but is it reasonable to expect autofit_columns to be more aware of the font in use and not assume it's the default one? Or is this the norm? Not knowing the tk libraries that much, I really don't know.

Thanks

**Comment by israel-dryer** [comment, marked as answer]
Great catch — this was a genuine limitation, not expected behavior. `autofit_columns` hardcoded `TkDefaultFont` for its `.measure()` calls, so any table using a different font (like your `style.configure(".", font=("Cascadia Code", 14))`) got columns sized for the wrong font and text ended up clipped. Your workaround was exactly the right idea.

I've fixed it in **#1158**: `autofit_columns` now measures with the font actually configured on the treeview style — cells use the `Table.Treeview` font and headers the `Table.Treeview.Heading` font — falling back to `TkDefaultFont` only when the style doesn't override it (so the default look is unchanged). Once that lands you won't need the manual measuring code; a plain `autofit=True` (or `table.autofit_columns()`) will respect your font.

Thanks for the clear report and the diagnosis!
